src/core/macro.scala

```diff
@@ -192,79 +211,323 @@ object Macro:
           .map(_.asExpr.asInstanceOf[Expr[Any]])
       }
 
-    def typeAnns: Expr[List[Any]] = {
-
-      def getAnnotations(t: TypeRepr): List[Term] = t match
-        case AnnotatedType(inner, ann) => ann :: getAnnotations(inner)
-        case _                         => Nil
-
+    def typeAnns: Expr[List[Any]] =
       val symbol: Option[Symbol] =
         if tpe.typeSymbol.isNoSymbol then None else Some(tpe.typeSymbol)
+
       Expr.ofList {
-        symbol.toList.map(_.tree).flatMap {
-          case ClassDef(_, _, parents, _, _) =>
-            parents
-              .collect { case t: TypeTree => t.tpe }
-              .flatMap(getAnnotations)
-              .filter(filterAnnotation)
-              .map(_.asExpr.asInstanceOf[Expr[Any]])
-          case _ =>
-            List.empty
-        }
+        symbol.toList
+          .map(_.tree)
+          .flatMap {
+            case ClassDef(_, _, parents, _, _) =>
+              parents
+                .collect {
+
+                  case t: TypeTree => t.tpe
+
+                  // case for AnyVal type annotations: with "-Yretain-trees" scalac option, the TypeTree of the annotation gets erased,
```

Comment:
with the option, or without the option? It would seem on first sight that trees get erased without `retain-trees` :)